daemon/cmd/daemon_main.go

```diff
@@ -350,6 +350,9 @@ func init() {
 	flags.Bool(option.EnableBPFTProxy, defaults.EnableBPFTProxy, "Enable BPF-based proxy redirection, if support available")
 	option.BindEnv(option.EnableBPFTProxy)
 
+	flags.Bool(option.EnableHostLegacyRouting, defaults.EnableHostLegacyRouting, "Enable the legacy host forwarding model which does not bypass upper stack in host namespace")
+	option.BindEnv(option.EnableHostLegacyRouting)
```

Comment:
I would even consider hiding this option. The way it works right now, if the kernel is the right version, we will automatically enable the optimization. If that turns out to be buggy, we can disable it from the CLI which is generally a nice safety valve. 

Exposing this to users just adds additional complexity which 99% cases will not need.

Reply:
I mean I left it there for those that would explicitly need iptables if they have their own additional rules as otherwise it's bypassed.

Reply:
I agree wrt your warning assessment. I think in that case, my preference would be to remove it altogether, and then add a cilium status field which can be dumped in verbose mode to have at least some way to introspect which mode we're in.